src/System.Net.Http/tests/FunctionalTests/DiagnosticsTests.cs

```diff
@@ -247,6 +280,324 @@ public void SendAsync_ExpectedDiagnosticCancelledLogging()
 
                 using (DiagnosticListener.AllListeners.Subscribe(diagnosticListenerObserver))
                 {
+                    diagnosticListenerObserver.Enable(s => !s.Contains("Activity"));
+                    using (var client = new HttpClient())
+                    {
+                        LoopbackServer.CreateServerAsync(async (server, url) =>
+                        {
+                            CancellationTokenSource tcs = new CancellationTokenSource();
+                            Task request = LoopbackServer.AcceptSocketAsync(server,
+                                (s, stream, reader, writer) =>
+                                {
+                                    tcs.Cancel();
+                                    return LoopbackServer.ReadWriteAcceptedAsync(s, reader, writer);
+                                });
+                            Task response = client.GetAsync(url, tcs.Token);
+                            await Assert.ThrowsAnyAsync<Exception>(() => Task.WhenAll(response, request));
+                        }).Wait();
+                    }
+                }
+                // Poll with a timeout since logging response is not synchronized with returning a response.
+                WaitForTrue(() => cancelLogged, TimeSpan.FromSeconds(1),
+                    "Cancellation was not logged within 1 second timeout.");
```

Comment:
This is a fairly short timeout... we've seen CI get bogged down and short timeouts like this end up causing spurious failures.  Would it hurt anything to extend this to, say, 10 seconds?